### PR TITLE
[FLINK-29310] Added license check

### PR DESCRIPTION
**Pull request by darenwkt**
Added license check as maven plugin to be used in ci workflow for license check.
Added Download Flink Binary to ci workflow to use flink-ci-tools package.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,7 +25,11 @@ jobs:
       matrix:
         jdk: [8, 11]
     env:
+      MVN_COMMON_OPTIONS: -U -B --no-transfer-progress
       MVN_CONNECTION_OPTIONS: -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false -Dmaven.wagon.httpconnectionManager.ttlSeconds=120
+      FLINK_URL: https://s3.amazonaws.com/flink-nightly/flink-1.15-SNAPSHOT-bin-scala_2.12.tgz
```
Comment (dannycranmer) [inline .github/workflows/ci.yml:30]: Why are we pointing at Flink 1.15-SNAPSHOT?

File: pom.xml [hunk 2]
```diff
@@ -83,6 +83,35 @@ under the License.
 
 	<build>
 		<plugins>
+			<plugin>
+				<groupId>org.codehaus.mojo</groupId>
+				<artifactId>exec-maven-plugin</artifactId>
+				<version>3.1.0</version>
+				<inherited>false</inherited>
```
Comment (dannycranmer) [inline pom.xml:90]: Can we inherit any of this, plus config, from the parent pom?

File: pom.xml [hunk 3]
```diff
@@ -83,6 +83,35 @@ under the License.
...
+				<configuration>
+					<mainClass>org.apache.flink.tools.ci.licensecheck.LicenseChecker</mainClass>
+					<includePluginDependencies>true</includePluginDependencies>
+					<includeProjectDependencies>false</includeProjectDependencies>
+				</configuration>
+				<dependencies>
+					<dependency>
+						<groupId>org.apache.flink</groupId>
+						<artifactId>flink-ci-tools</artifactId>
+						<version>1.15-SNAPSHOT</version>
```
Comment (dannycranmer) [inline pom.xml:110]: Can we not use the Flink parent version here?

**Review by dannycranmer — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (darenwkt) [reply .github/workflows/ci.yml:30]: Will use official Flink 1.15.2 at https://dlcdn.apache.org/flink/flink-1.15.2/flink-1.15.2-bin-scala_2.12.tgz

File: pom.xml (hunk 2 above)
Comment (darenwkt) [reply pom.xml:90]: Resolved in offline discussion with Danny

**Comment by darenwkt** [comment]
Moving this PR to: https://github.com/apache/flink-connector-dynamodb/pull/10 to pull from a different branch

**Pull request closed without merging**